app/src/main/java/com/arctouch/codechallenge/api/MoviesRepository.kt

```diff
@@ -0,0 +1,34 @@
+package com.arctouch.codechallenge.api
+
+import android.annotation.SuppressLint
+import com.arctouch.codechallenge.model.GenreResponse
+import com.arctouch.codechallenge.model.UpcomingMoviesResponse
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+
+//repository to centralize data request
+class MoviesRepository (private val api: TmdbApi): MoviesDataSource {
+    @SuppressLint("CheckResult")
+    override fun getMovieList(page: Long, success: (UpcomingMoviesResponse) -> Unit, failure: (Throwable) -> Unit) {
+        api.upcomingMovies(TmdbApi.API_KEY, TmdbApi.DEFAULT_LANGUAGE, page, TmdbApi.DEFAULT_REGION)
```

Comment:
Currently you're passing the API_KEY in every request. Can you think of a way to avoid this?

Reply:
In my current project I'm using a retrofit interceptor to add common properties in all the calls, like language, content-type, authorization, etc.